### DISCUSSION TITLE
[New Component] [ip] 能出个ip组件吗

### DISCUSSION THREAD

**Discussion by wtjperi2003** [element-plus/element-plus#14380] [Feature Request]
Feature type: **`New Component`**

## Details

- New component name: `el-ip`

## Description

可支持ip和端口的输入，也能只显示ip输入，就是不需要端口的时候用

### Additional comments

_(empty)_

**Comment by xxholly32** [comment]
自己写个就好咯

[ipinput](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQge3JlZn0gZnJvbSAndnVlJ1xuaW1wb3J0IElwSW5wdXQgZnJvbSAnLi9JcElucHV0LnZ1ZSc7XG5cbmNvbnN0IGlwID0gcmVmKCcnKVxuXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8SXBJbnB1dCBcbiAgICAgIHYtbW9kZWw9XCJpcFwiXG4gICAgICAvPlxuPC90ZW1wbGF0ZT4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl1cbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJzcmMvSXBJbnB1dC52dWUiOiI8c2NyaXB0IHNldHVwPlxyXG5pbXBvcnQgeyByZWYsd2F0Y2gscmVhY3RpdmUsb25Nb3VudGVkLG9uQmVmb3JlVW5tb3VudCB9IGZyb20gJ3Z1ZSdcclxuY29uc3QgcHJvcHMgPSBkZWZpbmVQcm9wcyh7XHJcbiAgaXBUaXBzOiB7XHJcbiAgICB0eXBlOiBTdHJpbmcsXHJcbiAgICBkZWZhdWx0OiAnJyxcclxuICB9LFxyXG59KVxyXG5jb25zdCBlbWl0ID0gZGVmaW5lRW1pdHMoWyd1cGRhdGU6bW9kZWxWYWx1ZSddKVxyXG5cclxuY29uc3QgaW5mbyA9IHJlZih7XHJcbiAgZmlyc3RJcDogJycsXHJcbiAgc2Vjb25kSXA6ICcnLFxyXG4gIHRoaXJkSXA6ICcnLFxyXG4gIGZvdXJ0aElwOiAnJyxcclxufSlcclxuXHJcbndhdGNoKGluZm8udmFsdWUsIChudikgPT4ge1xyXG4gIGlmKG52LmZpcnN0SXAgIT09ICcnICYmIG52LnNlY29uZElwICE9PSAnJyAmJiBudi50aGlyZElwICE9PSAnJyAmJiBudi5mb3VydGhJcCAhPT0gJycpe1xyXG4gICAgY29uc29sZS5sb2coJ2lwQ2hhbmdlJywgYCR7bnYuZmlyc3RJcH0uJHtudi5zZWNvbmRJcH0uJHtudi50aGlyZElwfS4ke252LmZvdXJ0aElwfWApXHJcbiAgICBlbWl0KCd1cGRhdGU6bW9kZWxWYWx1ZScsIGAke252LmZpcnN0SXB9LiR7bnYuc2Vjb25kSXB9LiR7bnYudGhpcmRJcH0uJHtudi5mb3VydGhJcH1gKVxyXG4gIH1cclxuICAgIFxyXG59KVxyXG5cclxuY29uc3QgaXBSZWYgPSByZWYobnVsbClcclxuY29uc3QgaXBMaXN0ID0gcmVhY3RpdmUoW3tcclxuICBrZXk6ICdmaXJzdElwJyxcclxufSwge1xyXG4gIGtleTogJ3NlY29uZElwJyxcclxufSwge1xyXG4gIGtleTogJ3RoaXJkSXAnLFxyXG59LCB7XHJcbiAga2V5OiAnZm91cnRoSXAnLFxyXG59XSlcclxuXHJcbi8vIOebkeWQrOeymOi0tFxyXG5mdW5jdGlvbiBoYW5kbGVQYXN0ZShldmVudCkge1xyXG4gIC8vIOWkhOeQhueymOi0tOS6i+S7tueahOmAu+i+kVxyXG4gIGNvbnN0IGNsaXBib2FyZERhdGEgPSBldmVudC5jbGlwYm9hcmREYXRhIHx8IHdpbmRvdy5jbGlwYm9hcmREYXRhXHJcbiAgY29uc3QgcGFzdGVkVGV4dCA9IGNsaXBib2FyZERhdGEuZ2V0RGF0YSgndGV4dC9wbGFpbicpXHJcblxyXG4gIGNvbnNvbGUubG9nKCdQYXN0ZWQgdGV4dDonLCBwYXN0ZWRUZXh0KVxyXG4gIGNvbnN0IGFyciA9IHBhc3RlZFRleHQuc3BsaXQoJy4nKVxyXG4gIGNvbnNvbGUubG9nKGFycilcclxuICBpZiAoYXJyLmxlbmd0aCA9PT0gNCkge1xyXG4gICAgLy8g6ZqU57ud6Z2e5bi46KeE5YGHaXBcclxuICAgIGlmIChhcnIuc29tZShpdGVtID0+IGlzTmFOKHBhcnNlSW50KGl0ZW0pKSkpIHtcclxuICAgICAgZXZlbnQucHJldmVudERlZmF1bHQoKVxyXG4gICAgICByZXR1cm5cclxuICAgIH1cclxuICAgIFtcclxuICAgICAgaW5mby52YWx1ZS5maXJzdElwLFxyXG4gICAgICBpbmZvLnZhbHVlLnNlY29uZElwLFxyXG4gICAgICBpbmZvLnZhbHVlLnRoaXJkSXAsXHJcbiAgICAgIGluZm8udmFsdWUuZm91cnRoSXAsXHJcbiAgICBdID0gYXJyXHJcbiAgICBldmVudC5wcmV2ZW50RGVmYXVsdCgpXHJcbiAgfVxyXG59XHJcblxyXG4vLyDojrflj5blvZPliY3ovpPlhaXmoYbnmoTnhKbngrnkvY3nva4gYnkgbGl1bWVuZzZcclxuY29uc3QgZ2V0UmFuZ2UgPSAoZWwpID0+IHtcclxuICBjb25zdCByZXQgPSB7fVxyXG4gIGlmIChlbC5zZXRTZWxlY3Rpb25SYW5nZSkge1xyXG4gICAgcmV0LmJlZ2luID0gZWwuc2VsZWN0aW9uU3RhcnRcclxuICAgIHJldC5lbmQgPSBlbC5zZWxlY3Rpb25FbmRcclxuICAgIHJldC5yZXN1bHQgPSBlbC52YWx1ZS5zdWJzdHJpbmcocmV0LmJlZ2luLCByZXQuZW5kKVxyXG4gIH1cclxuICBlbC5mb2N1cygpXHJcbiAgcmV0dXJuIHJldFxyXG59XHJcblxyXG4vKipcclxuICAgICogQGF1dGhvciBsaXVtZW5nNlxyXG4gICAgKiBAZGF0ZSAyMDE3LTEyLTExIDE1OjUwOjQ4XHJcbiAgICAqIEBkZXNjIGlucHV05oyJ6ZSu55uR5ZCsXHJcbiAgICAqL1xyXG5jb25zdCBvbklucHV0S2V5dXAgPSAoZXZlbnQsIGluZGV4KSA9PiB7XHJcbiAgY29uc3Qga2V5Q29kZSA9IGV2ZW50LmtleUNvZGUgfHwgZXZlbnQud2hpY2hcclxuICBjb25zdCB2YWx1ZSA9IGV2ZW50LnRhcmdldC52YWx1ZVxyXG4gIGNvbnN0IGFyckluZGV4ID0gaW5kZXggLy8g5YC85Zyo5pWw57uE5Lit55qE6aG65bqPXHJcblxyXG4gIGNvbnNvbGUubG9nKGluZGV4KVxyXG4gIGNvbnNvbGUubG9nKGlwUmVmKVxyXG4gIGNvbnN0IGlzTmVlZENvZGUgPSAvXlswLTkuXSskLy50ZXN0KHZhbHVlKVxyXG4gIGNvbnN0IGZpcnRzdEluZGV4ID0gdmFsdWUuaW5kZXhPZignMCcpXHJcbiAgLy8g6Iux5paH5a2X5Lul5Y+K5YW25LuW5a2X56ym5YiZ5Yig6Zmk5o6JXHJcbiAgaWYgKCFpc05lZWRDb2RlICYmIGtleUNvZGUgIT09IDggJiYga2V5Q29kZSAhPT0gMzcgJiYga2V5Q29kZSAhPT0gMzkpIHtcclxuICAgIGluZm8udmFsdWVbaXBMaXN0W2FyckluZGV4XS5rZXldID0gdmFsdWUuc2xpY2UoMCwgdmFsdWUubGVuZ3RoIC0gMSlcclxuICAgIHJldHVyblxyXG4gIH1cclxuICBpZiAoZmlydHN0SW5kZXggPT09IDApIHtcclxuICAgIGluZm8udmFsdWVbaXBMaXN0W2FyckluZGV4XS5rZXldID0gdmFsdWUuc2xpY2UoMCwgMSlcclxuICAgIHJldHVyblxyXG4gIH1cclxuXHJcbiAgLy8g5aSE55CG6ZSu55uY55uR5ZCs77yMOCDkuLrpgIDmoLzplK7vvIwzNyDkuLpsZWZ0566t5aS077yMMzkg5Li6cmlnaHTnrq3lpLTvvIwxMTAg5Li65pWw5a2X6ZSu55uY5LiK55qE5bCP5pWw54K577yMMTkw5Li65o6n5Yi26ZSu5LiK55qE5bCP5pWw54K577yMMjI55Li65Lit5paH5pWw5a2X6ZSu55uY5LiK5bCP5pWw54K5XHJcbiAgaWYgKGtleUNvZGUgPT09IDggfHwga2V5Q29kZSA9PT0gMzcpIHtcclxuICAgIGlmIChcclxuICAgICAgKHZhbHVlLmxlbmd0aCA9PT0gMCB8fCBnZXRSYW5nZShldmVudC50YXJnZXQpLmVuZCA9PT0gMClcclxuICAgICAgICAgICAgJiYgYXJySW5kZXggPiAwXHJcbiAgICApXHJcbiAgICBpcFJlZi52YWx1ZVthcnJJbmRleCAtIDFdLmZvY3VzKClcclxuICB9XHJcbiAgZWxzZSBpZiAoa2V5Q29kZSA9PT0gMzkpIHtcclxuICAgIC8vIOaMieS4i+WPs+eureWktO+8jOimgeWFieagh+WcqOacgOWQjueahOaXtuWAmeaJjeiBmueEpuS4i+S4gOS4qlxyXG4gICAgaWYgKGdldFJhbmdlKGV2ZW50LnRhcmdldCkuZW5kICE9PSB2YWx1ZS5sZW5ndGgpXHJcbiAgICAgIHJldHVyblxyXG4gICAgaWYgKGFyckluZGV4IDwgMylcclxuICAgIGlwUmVmLnZhbHVlW2FyckluZGV4ICsgMV0uZm9jdXMoKVxyXG4gIH1cclxuICBlbHNlIGlmIChcclxuICAgIGtleUNvZGUgPT09IDExMFxyXG4gICAgICAgIHx8IGtleUNvZGUgPT09IDE5MFxyXG4gICAgICAgIHx8IChrZXlDb2RlID09PSAyMjlcclxuICAgICAgICAgICAgJiYgKGV2ZW50LmNvZGUgPT09ICdOdW1wYWREZWNpbWFsJyB8fCBldmVudC5jb2RlID09PSAnUGVyaW9kJykpXHJcbiAgKSB7XHJcbiAgICBjb25zb2xlLmxvZyhpcFJlZi52YWx1ZVthcnJJbmRleCArIDFdKVxyXG4gICAgaWYgKCF2YWx1ZS5sZW5ndGgpXHJcbiAgICAgIHJldHVyblxyXG4gICAgY29uc3QgX251bSA9IHZhbHVlW3ZhbHVlLmxlbmd0aCAtIDFdID09PSAnLicgPyAodmFsdWUubGVuZ3RoIC0gMSkgOiAzXHJcbiAgICBpbmZvLnZhbHVlW2lwTGlzdFthcnJJbmRleF0ua2V5XSA9IHZhbHVlLnNsaWNlKDAsIF9udW0pXHJcbiAgICAvLyDovpPlhaXlsI/mlbDngrnmiJbmlbDlrZfplK7nm5jkuIrnmoTlj6Xlj7fog73ot7PovaxcclxuICAgIGlmIChhcnJJbmRleCA8IDMpXHJcbiAgICBpcFJlZi52YWx1ZVthcnJJbmRleCArIDFdLmZvY3VzKClcclxuICB9XHJcbn1cclxuXHJcbm9uTW91bnRlZCgoKSA9PiB7XHJcbiAgY29uc3QgZG9tID0gZG9jdW1lbnQucXVlcnlTZWxlY3RvcignI3Bhc3REb20nKVxyXG4gIC8vIOa3u+WKoOeymOi0tOS6i+S7tueahOebkeWQrOWZqFxyXG4gIGRvbS5hZGRFdmVudExpc3RlbmVyKCdwYXN0ZScsIGhhbmRsZVBhc3RlKVxyXG5cclxufSlcclxub25CZWZvcmVVbm1vdW50KCgpID0+IHtcclxuICBjb25zdCBkb20gPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yKCcjcGFzdERvbScpXHJcbiAgZG9tLnJlbW92ZUV2ZW50TGlzdGVuZXIoJ3Bhc3RlJywgaGFuZGxlUGFzdGUpXHJcbn0pXHJcbjwvc2NyaXB0PlxyXG5cclxuPHRlbXBsYXRlPlxyXG4gIDxkaXYgY2xhc3M9XCJpcENudCB3LWZ1bGwgZmxleCBpdGVtcy1jZW50ZXJcIj5cclxuICAgIFxyXG4gICAgPGRpdiBjbGFzcz1cImJveCBoLTMycHggZmxleCBmbGV4LTEgZWwtaW5wdXRfX3dyYXBwZXIgYm94LWJvcmRlclwiPlxyXG4gICAgICAgIDxkaXYgaWQ9XCJwYXN0RG9tXCIgY2xhc3M9XCJmbGV4IGgtZnVsbFwiPlxyXG4gICAgICAgICAgPHRlbXBsYXRlIHYtZm9yPVwiKGl0ZW0sIGluZGV4KSBpbiBpcExpc3RcIiA6a2V5PVwiaW5kZXhcIj5cclxuICAgICAgICAgICAgPGVsLWlucHV0XHJcbiAgICAgICAgICAgICAgcmVmPVwiaXBSZWZcIiB2LW1vZGVsLnRyaW09XCJpbmZvW2l0ZW0ua2V5XVwiIDptYXhsZW5ndGg9XCIzXCJcclxuICAgICAgICAgICAgICBAa2V5dXA9XCJvbklucHV0S2V5dXAoJGV2ZW50LCBpbmRleClcIlxyXG4gICAgICAgICAgICAvPlxyXG4gICAgICAgICAgICA8dGVtcGxhdGUgdi1pZj1cImluZGV4IDwgaXBMaXN0Lmxlbmd0aCAtIDFcIj5cclxuICAgICAgICAgICAgICAuXHJcbiAgICAgICAgICAgIDwvdGVtcGxhdGU+XHJcbiAgICAgICAgICA8L3RlbXBsYXRlPlxyXG4gICAgICAgIDwvZGl2PlxyXG4gICAgPC9kaXY+XHJcbiAgPC9kaXY+XHJcbjwvdGVtcGxhdGU+XHJcblxyXG48c3R5bGUgc2NvcGVkPlxyXG4uaXBDbnQge1xyXG4gICAgd2lkdGg6IDEwMCU7XHJcbiAgICBkaXNwbGF5OiBmbGV4O1xyXG4gICAgYWxpZ24taXRlbXM6IGNlbnRlcjtcclxufVxyXG4uaXBDbnQgPiBzcGFue1xyXG4gICAgcGFkZGluZzogMCA1cHg7XHJcbn1cclxuLmJveCB7XHJcbiAgICBwYWRkaW5nLWxlZnQ6IDFweDtcclxuICAgIHBhZGRpbmctcmlnaHQ6IDJweDtcclxuICAgIGZsZXg6IDE7XHJcbiAgICBkaXNwbGF5OiBmbGV4O1xyXG4gICAgaGVpZ2h0OiAzMnB4O1xyXG4gICAgYm94LXNpemluZzogYm9yZGVyLWJveDtcclxuICAgIGJveC1zaGFkb3c6IDAgMCAwIDFweCAjYjNiM2IzIGluc2V0O1xyXG59XHJcbi5ib3g6aG92ZXJ7XHJcbiAgICBib3gtc2hhZG93OiAwIDAgMCAxcHggIzMzMyBpbnNldDtcclxufVxyXG4uYm94ID4gZGl2e1xyXG4gICAgZGlzcGxheTogZmxleDtcclxuICAgIGJveC1zaXppbmc6IGJvcmRlci1ib3g7XHJcbiAgICBoZWlnaHQ6IDEwMCU7XHJcbn1cclxuXHJcbjo6di1kZWVwIC5ib3ggLmVsLWlucHV0X193cmFwcGVyIHtcclxuICAgIGJveC1zaGFkb3c6IG5vbmUgIWltcG9ydGFudDtcclxuICAgIHBhZGRpbmctbGVmdDogMnB4O1xyXG4gICAgcGFkZGluZy1yaWdodDogMnB4O1xyXG59XHJcbjo6di1kZWVwIC5ib3ggLmVsLWlucHV0X193cmFwcGVyIGlucHV0IHtcclxuICAgIHRleHQtYWxpZ246IGNlbnRlcjtcclxufVxyXG5cclxuLyog5ZyoQ2hyb21l5rWP6KeI5Zmo5LiLICovXHJcbjo6di1kZWVwIC5ib3ggLmVsLWlucHV0X193cmFwcGVyIGlucHV0Ojotd2Via2l0LW91dGVyLXNwaW4tYnV0dG9uLFxyXG46OnYtZGVlcCAuYm94IC5lbC1pbnB1dF9fd3JhcHBlciBpbnB1dDo6LXdlYmtpdC1pbm5lci1zcGluLWJ1dHRvbiB7XHJcbiAgICAtd2Via2l0LWFwcGVhcmFuY2U6IG5vbmU7XHJcbn1cclxuXHJcbi8qIOWcqEZpcmVmb3jmtY/op4jlmajkuIsgKi9cclxuOjp2LWRlZXAgLmJveCAuZWwtaW5wdXRfX3dyYXBwZXIgaW5wdXRbdHlwZT1cIm51bWJlclwiXSB7XHJcbiAgICAtbW96LWFwcGVhcmFuY2U6IHRleHRmaWVsZDtcclxufVxyXG5cclxuXHJcbjo6di1kZWVwIC5ib3ggIC5lbC1pbnB1dC1udW1iZXIge1xyXG4gICAgd2lkdGg6IGF1dG87XHJcbn1cclxuXHJcbjo6di1kZWVwIC5ib3ggOjpob3ZlciB7XHJcbiAgICBib3gtc2hhZG93OiAwIDAgMCAxcHggIzMzMyBpbnNldDtcclxufVxyXG5cclxuPC9zdHlsZT5cclxuIiwiX28iOnt9fQ==)

**Reply by wtjperi2003** [reply]
兄弟，不是写不来，是希望能在el组件库里，像暗黑主题，国际化等一脉相承。
